main.js

```diff
@@ -329,9 +344,16 @@ function plotStock(graphName, ticker, tickerString, color, forceColor, scaleInde
     .defined(function(d) {
       return d != null;
     });
+  
+  // for the 1D and 5D graphs, the stock data will not always extend
+  // to the end of the x-axis (aka midday, the plot will fill half the graph).
+  var numBufferPoints = Math.floor(xScaleDomain - numDataDates);
+  var endDateForScale = new Date(plotData.dates[numDataDates - 1]).getTime() + numBufferPoints*5*60*1000/plotData.time.period;
```

Comment:
is this really easier than just hardcoding the end of the day, which is always 16:00?